internals/overlord/restart/restart.go

```diff
@@ -149,24 +150,20 @@ func Request(st *state.State, t RestartType) {
 	case RestartSystem, RestartSystemNow, RestartSystemHaltNow, RestartSystemPoweroffNow:
 		st.Set("system-restart-from-boot-id", rm.bootID)
 	}
-	rm.restarting = t
+	rm.restarting.Store(int32(t))
 	rm.handleRestart(t)
 }
 
 // Pending returns whether a restart was requested with Request and of which type.
-func Pending(st *state.State) (bool, RestartType) {
-	cached := st.Cached(restartManagerKey{})
-	if cached == nil {
-		return false, RestartUnset
-	}
-	rm := cached.(*RestartManager)
-	return rm.restarting != RestartUnset, rm.restarting
+// NOTE: the state does not need to be locked to fetch this information.
+func (rm *RestartManager) Pending() (bool, RestartType) {
```

Comment:
an "ok" bool is usually the last return value (like error)

e.g.
`v, ok := myMap[k]`

Since this is existing functionality, I don't mind.

Reply:
Yeah, I was actually tempted to get rid of the bool entirely, because it's just `b != RestartUnset`, which is easy for the caller to do. However, given it has this signature in snapd, I want to keep it as is.